x/authority/keeper/msg_server_add_authorization.go

```diff
@@ -0,0 +1,38 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/zeta-chain/zetacore/x/authority/types"
+)
+
+// AddAuthorization defines a method to add an authorization.If the authorization already exists, it will be overwritten with the provided policy.
+// This should be called by the admin policy account.
+func (k msgServer) AddAuthorization(
+	goCtx context.Context,
+	msg *types.MsgAddAuthorization,
+) (*types.MsgAddAuthorizationResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if !k.IsAuthorized(ctx, msg.Creator, types.PolicyType_groupAdmin) {
+		return nil, errorsmod.Wrap(
+			types.ErrUnauthorized,
+			"AddAuthorization can only be executed by the admin policy account",
+		)
+	}
+
+	authorizationList, _ := k.GetAuthorizationList(ctx)
+	authorizationList.SetAuthorization(types.Authorization{MsgUrl: msg.MsgUrl, AuthorizedPolicy: msg.AuthorizedPolicy})
+
+	// validate the authorization list after adding the authorization as a precautionary measure.
+	err := authorizationList.Validate()
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "authorization list is invalid")
+	}
+
+	k.SetAuthorizationList(ctx, authorizationList)
+	return &types.MsgAddAuthorizationResponse{}, nil
```

Comment:
Choose not to emit an event , as this message would be submitted by the policy owners and might not need tobe tracked via an event